File: app/models/event.rb
class Event < ApplicationRecord
  belongs_to :location, class_name: 'EventLocation'
  has_many :event_sign_ups
  validates_presence_of :name, :description, :location
  validates_datetime :starts
  validates_datetime :ends
  validates_datetime :ends, :after => :starts, if: Proc.new {|o| !o.starts.blank? and !o.ends.blank? }
end

File: app/javascript/reducers/event.js
import { FETCH_EVENT } from '../actions/types';

const initialState = {
    id: '',
    name: '',
    description: '',
    location: {},
    starts: '',
    ends: ''
};

export default function(state = initialState, action) {
    switch (action.type) {
        case FETCH_EVENT:
            return {
                ...state,
                ...action.payload
            };
        default:
            return state;
    }
}